src/osd/osd_types.h

```diff
@@ -438,15 +434,15 @@ struct pg_t {
     encode(v, bl);
     encode(m_pool, bl);
     encode(m_seed, bl);
-    encode(m_preferred, bl);
+    encode((int32_t)-1, bl);
```

Comment:
Please comment it here. How about

> // previously held a pg preferred value

Or we could just have a named "int32_t reserved = -1;" sitting there.